README.md

```diff
@@ -1,58 +1,195 @@
+- `/root/BUILD/rpms/x86_64/manageis-*.rpm`
 
-4. Run the script
+Depending upon your usecase, there are a few ways to obtain these artifacts:
 
-   - Run `bin/build.rb`, optionally overriding git ref (branch or tag) and release type
+- Have the container put the artifacts into a mounted volume.
+- Manually copy the artifacts to a mounted volume.
+- Extracting the artifacts from the docker container with a docker command.
 
-     ```
-     bin/build.rb --git-ref jansa --update-rpm-repo
-     ```
-     ```
-     bin/build.rb --build-type release --git-ref jansa-1-alpha1 --update-rpm-repo
-     ```
+### Having the container build files into a mounted volume
 
-Alternatively, the build can be started directly by passing `build` to the
-`docker run` command:
+Since the files are not uploaded, they need to be retrieved out of the repo before the container exits.
 
+Note: all mounted volumes (i.e.: using the `-v` option) need to be defined in docker's preferences:
+
+```
+Settings > Resources > FILE SHARING
+```
+
+For this example, 2 directories exist in the current directory (`pwd`):
+
+- `OPTIONS` a directory contains the `options.yml` file
+- `BUILD` a directory that will contain the `tar.gz` and `rpm` files.
+
+Note: The contents of `BUILD` will be deleted before each run.
+
+```sh
+docker run -v --rm -v `pwd`/OPTIONS:/root/OPTIONS -v `pwd`/BUILD:/root/BUILD manageiq/rpm_build:latest build
+```
+
+### Manually copying files to a mounted volume
+
+Since defining the mount points can be inconvenient, it is possible to just run a build and copy out the files out.
+The trick is to run the container as a daemon and in a way that it does not exit before you have copied out the resources.
+
+#### Running the container in daemon mode
+
+```sh
+# setup docker and remember the CONTAINER id
+CONTAINER=`docker run -d -it -v $(pwd)/OPTIONS:/root/OPTIONS manageiq/rpm_build:latest`
+```
+
+Alternatively `CONTAINER` can be assigned from the list of recent docker containers:
+
+```sh
+docker container ls
+```
+
+#### Attaching and running a build
+
+The commands to run a build in the docker container can be found in the [entrypoint](blob/master/container-assets/user-entrypoint.sh#L17) script.
+
+```sh
+docker exec -it ${CONTAINER} /bin/bash
+
+cd /build_scripts
+export GEM_HOME=$(ruby -e 'puts Gem.user_dir')
+export PATH=$GEM_HOME/bin:$HOME/bin:$PATH
+bundle
+./bin/build.rb
+```
+
+#### Pushing the build artifacts to a mounted volume
+
+From here you can copy the rpms to an output. `/root/OPTIONS/rpms/` is convenient since `/root/OPTIONS` is mounted anyway.
+
+```sh
+cp -v /root/BUILD/rpms/x86_64/m* /root/OPTIONS/rpms/
 ```
-docker run -it <miq-rpm_build image> build [--build-type <type>] [--git-ref <ref>] [--update-rpm-repo]
+
+#### Pulling the build artifacts out of docker
+
+An alternative is to exit out of docker and have docker go into the container and copy them to your local file system
+
+```sh
+docker cp ${CONTAINER}:/root/BUILD/rpms/x86_64/ ./rpms/
```

Comment:
This is generally the easiest way if you're using a daemonized container...I think we should just remove the "Pushing the build artifacts to a mounted volume" section, personally, to simplify (too many choices => paradox of choice)

Reply:
to be honest, I tend to go in, build it, an while I'm in there just copy to /root/OPTIONS

I'd prefer just a `--copy-rpms $TARGET` option using build and be done with it.

Reply:
> to be honest, I tend to go in, build it, an while I'm in there just copy to /root/OPTIONS
> 
> I'd prefer just a `--copy-rpms $TARGET` option using build and be done with it.

Wouldn't it be easier to use a second volume mount rather than daemonize, exec into it, then copy to a volume mount?

Reply:
1. removing push
2. I find it easier to mount all these directories but I will try again and see how it goes. I've already been convinced of a number of changes that are making this easier

Reply:
thanks for the question. Misunderstandings of previous errors had me going in a tangent.